lib/api/core/indexCache.js

```diff
@@ -48,58 +49,48 @@ class IndexCache {
   }
 
   init () {
-    const engine = this.kuzzle.internalEngine;
-    const kuzzleInfoFieldsRegex = new RegExp(`^_kuzzle_info\\.(${Object.keys(this.commonMapping._kuzzle_info.properties).join('|')})$`);
+    const internalEngine = this.kuzzle.internalEngine;
 
-    return engine.listIndexes()
+    return internalEngine.listIndexes()
       .then(indexes => indexes.filter(index => index !== this.kuzzle.internalEngine.index))
       .then(indexes => {
         const promises = [];
 
-        for (const index of indexes.filter(idx => idx !== this.kuzzle.internalEngine.index)) {
+        for (const index of indexes) {
           this.indexes[index] = [];
           this.defaultMappings[index] = _.cloneDeep(this.commonMapping);
 
-          promises.push(engine.getFieldMapping({
-            index,
-            fields: '_kuzzle_info.*'
-          })
-            .then(kuzInfoMappings => {
-              if (kuzInfoMappings[index] && kuzInfoMappings[index].mappings) {
-                for (const collection of Object.keys(kuzInfoMappings[index].mappings)) {
-                  const collectionInfo = kuzInfoMappings[index].mappings[collection];
-
-                  for (const dotField of Object.keys(collectionInfo).filter(f => kuzzleInfoFieldsRegex.test(f))) {
-                    Object.assign(this.defaultMappings[index]._kuzzle_info.properties, collectionInfo[dotField].mapping);
-                  }
-                  // do not break to catch cases where not all fields are defined
-                }
-              }
-
-              return engine.listCollections(index);
-            })
-            .then(collections => {
-              const updateMappingPromises = [];
+          promises.push(
+            internalEngine.listCollections(index)
+              .then(collections => {
+                const promisesAgain = [];
```

Comment:
I think we should make an effort to find a better name for the `promises` and `promisesAgain` arrays. Here are my ideas
* `promises` => `loopOnCollectionsPromises`
* `promisesAgain` => `applyDefaultMappingPromises`

Reply:
I know this is an issue that we will be wiped away by `async` `await`, but still...

Reply:
Nope. We shouldn't use async/await in the index cache, not until we make sure there aren't performance issues with this new feature: AFAIK there are I-don't-remember-how-many promises created for each async function, which may be a problem under pressure because this can clog up the event loop pretty fast.